packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1073,6 +1074,48 @@ export function attach(
     return null;
   }
 
+  function areHookInputsEqual(
+    nextDeps: Array<mixed>,
+    prevDeps: Array<mixed> | null,
+  ) {
+    if (prevDeps === null) {
+      return false;
+    }
+
+    for (let i = 0; i < prevDeps.length && i < nextDeps.length; i++) {
+      if (is(nextDeps[i], prevDeps[i])) {
+        continue;
+      }
+      return false;
+    }
+    return true;
+  }
+
+  function isEffect(memoizedState) {
+    return (
+      memoizedState !== null &&
+      typeof memoizedState === 'object' &&
+      memoizedState.hasOwnProperty('tag') &&
+      memoizedState.hasOwnProperty('create') &&
+      memoizedState.hasOwnProperty('destroy') &&
+      memoizedState.hasOwnProperty('deps') &&
+      memoizedState.hasOwnProperty('next')
+    );
+  }
+
+  function didHookChange(prev: any, next: any): boolean {
+    const prevMemoizedState = prev.memoizedState;
+    const nextMemoizedState = next.memoizedState;
+
+    if (isEffect(prevMemoizedState) && isEffect(nextMemoizedState)) {
```

Comment:
How would the return values for `isEffect(prevMemoizedState)` and `isEffect(nextMemoizedState)` ever differ?

Reply:
I think so. But I haven't checked if Effect hooks start with `memoizedState = null`. I'll see if can improve the type story here.

Reply:
If they did, that wouldn't matter here because we early return in `didHooksChange` if either one was null.

Reply:
Let's just do this tweak in a follow up? :) Would be nice.